modules/solid_mechanics/src/materials/ConstitutiveModel.C

```diff
@@ -93,13 +95,19 @@ ConstitutiveModel::applyThermalStrain(unsigned qp,
                                       SymmTensor & strain_increment,
                                       SymmTensor & d_strain_dT)
 {
-  if (_has_temp && _t_step != 0)
+  if (_t_step == 1)
```

Comment:
This PR looks fine but in terms of defensible programming I would generally prefer to see `_t_step >= 1` instead of the current line in case the linear behavior of `_t_step` is changed in the future.